### PR TITLE
konlpy 0.6.0 ❄️

### PR DESCRIPTION
**Pull request by msentissi**
konlpy 0.6.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4752](https://anaconda.atlassian.net/browse/PKG-4752) 
- [Upstream repository](https://github.com/konlpy/konlpy/tree/v0.6.0)

### Explanation of changes:

- Skipped linux-s390x as there is no openjdk built for it.
- Skipped win-64 as it requires additional effort and is a nice to have.
- Skipped Mecab-Ko and Mecab-Ko-Dic as it is an optional dependency for which we have no packages ready. And the packages on conda-forge are platform-limited.
- Excluded python 3.12 as it breaks the build and is a nice to have.

[PKG-4752]: https://anaconda.atlassian.net/browse/PKG-4752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ryanskeith — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,45 +1,54 @@
...
 
 test:
   imports:
-    - {{ name.replace('-', '.') }}
-  requires:
-    - pip
+    - konlpy
   commands:
     - pip check
+    - pytest -k "(not test_mecab.py)"
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
    # ...
    - pytest -vv --ignore=test/test_mecab.py
```

could you please add a comment here?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: <sha256>
+  url: https://github.com/{{ name }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 1b576aef07d094706e4bae446ab835855abbe806647fe861243c378482eb79c6
 
 build:
-  number: 0
+  skip: True                                    # [py<38 or py>=312 or s390x or win]
```
Comment (lorepirri) [inline recipe/meta.yaml:15]: could you please add a comment about win? which additional effort is required? missing deps?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,45 +1,54 @@
...
-  summary: <PLEASE ADD SUMMARY>
-  description: |
-    <PLEASE ADD DESCRIPTION>
-  license: <PLEASE ADD LICENSE>
-  license_family: <PLEASE ADD LICENSE_FAMILY>
-  license_file: <PLEASE_ADD_LICENSE_FILE>
-  dev_url: <PLEASE ADD DEV URL>
-  doc_url: <PLEASE ADD DOC URL>
+  home: http://konlpy.org
+  doc_url: https://konlpy.org/en/latest/
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://konlpy.org
```

no language in the urls

**Comment by lorepirri** [comment]
@msentissi how about this comment https://anaconda.atlassian.net/browse/PKG-4752?focusedCommentId=221029

**Comment by msentissi** [comment]
> @msentissi how about this comment https://anaconda.atlassian.net/browse/PKG-4752?focusedCommentId=221029

@boldorider4 communicated the issue to SF and we got the green light to continue work without patching the download functionality out.

**Comment by lorepirri** [comment]
> > @msentissi how about this comment https://anaconda.atlassian.net/browse/PKG-4752?focusedCommentId=221029
> 
> @boldorider4 communicated the issue to SF and we got the green light to continue work without patching the download functionality out.

oh I now understand the comment in the ticket, thanks! Apologies, I did not read it properly.

**Comment by lorepirri** [comment]
@msentissi 
`win` fails with:
```
INFO       FileNotFoundError: [Errno 2] No such file or directory: 'C:\\b\\abs_1fodhrwz2t\\croot\\tmppdwdovy7\\konlpy-0.6.0\\CONTRIBUTING-en.rst ' 
```
~sometimes tests require the README or some other file, dunno, they test their own content maybe.~ in this case it's used in the setup

See conversation: https://anaconda.slack.com/archives/C02K52E033M/p1728406414285799

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,45 +1,55 @@
...
+    - konlpy
   commands:
     - pip check
+    # Ignoring Mecab-Ko testing because it's an optional dependency that is not included and not packaged
+    - pytest -vv --ignore=test/test_mecab.py
+  requires:
+    - pip
+    - pytest
+  source_files:
+    - test
```
Comment (lorepirri) [inline recipe/meta.yaml:43]: forget the previous commment, `CONTRIBUTING-en.rst` is required at setup

**Comment by lorepirri** [comment]
Running the tests for `win` will lead to this: https://github.com/konlpy/konlpy/issues/275

It seems that every time there is a `jvm.init_jvm`, it does not go well.

Install on win: https://konlpy.org/en/latest/install/#id2

If this gets too much of your time we could skip :)

**Comment by msentissi** [comment]
> Running the tests for `win` will lead to this: [konlpy/konlpy#275](https://github.com/konlpy/konlpy/issues/275)
> 
> It seems that every time there is a `jvm.init_jvm`, it does not go well.
> 
> Install on win: https://konlpy.org/en/latest/install/#id2
> 
> If this gets too much of your time we could skip :)

Regarding the original issue, I tried to patch it out, but the error appears to crash conda-build before it gets a chance to apply the patch. For the failures you expect, most of them would be fine since we skip the Mecab tests, but there is that one test_komoran.py failure which I find rather concerning.
Given these facts and the time-constraints, and given that Windows is a nice to have, I would lean towards leaving windows out.

**Comment by lorepirri** [comment]
> > Running the tests for `win` will lead to this: [konlpy/konlpy#275](https://github.com/konlpy/konlpy/issues/275)
> > It seems that every time there is a `jvm.init_jvm`, it does not go well.
> > Install on win: https://konlpy.org/en/latest/install/#id2
> > If this gets too much of your time we could skip :)
> 
> Regarding the original issue, I tried to patch it out, but the error appears to crash conda-build before it gets a chance to apply the patch. For the failures you expect, most of them would be fine since we skip the Mecab tests, but there is that one test_komoran.py failure which I find rather concerning. Given these facts and the time-constraints, and given that Windows is a nice to have, I would lean towards leaving windows out.

agree @msentissi , let's skip if it takes more than expected.

using a release file from the commit that fixes the issue (one more actually):

```
{% set commit_id = "e858682feb2a3ac73280af94bc234f0282b8723b" %}

...

source:
  url: https://github.com/{{ name }}/{{ name }}/archive/{{ commit_id }}.tar.gz
  sha256: e40e44d482a7fa09b11ea29140807813b15f30d30a376fce9d3b3da3dec64326
```

and ignoring these test files makes it work, but not so many tests run
```
    - pytest -vv --ignore=test/test_mecab.py --ignore=test/test_hannanum.py --ignore=test/test_kkma.py --ignore=test/test_komoran.py --ignore=test/test_openkoreantext.py
```

it seems cumbersome and the source files would not be the same as for v0.6.0 on pypi.

I would skip it.

File: recipe/meta.yaml (hunk 2 above)
Comment (msentissi) [reply recipe/meta.yaml:15]: Done ✅

**Comment by lorepirri** [comment]
A concern worth addressing, while working on this I can see that there are jar files that are binaries:
```
  adding 'konlpy/java/aho-corasick.jar'
  adding 'konlpy/java/jhannanum-0.8.4.jar'
  adding 'konlpy/java/kkma-2.0.jar'
  adding 'konlpy/java/komoran-3.0.jar'
  adding 'konlpy/java/open-korean-text-2.1.0.jar'
  adding 'konlpy/java/scala-library-2.12.3.jar'
  adding 'konlpy/java/shineware-common-1.0.jar'
  adding 'konlpy/java/shineware-ds-1.0.jar'
  adding 'konlpy/java/snakeyaml-1.12.jar'
  adding 'konlpy/java/twitter-text-1.14.7.jar'
```
@JeanChristopheMorinPerso can I ask your opinion on this? Shall we leave them there. They are probably the object of this package (JVM on, jar run).

https://github.com/konlpy/konlpy/tree/v0.6.0/konlpy/java

**Comment by JeanChristopheMorinPerso** [comment]
@lorepirri They don't provide any info on where they got the jars from. SOMe of them kind of seem related to https://github.com/koalanlp/koalanlp but were committed 10 years ago which doesn't match with what I found on [maven central](https://central.sonatype.com/).

The only thing we could do is compile https://github.com/konlpy/konlpy/tree/v0.6.0/konlpy/java/src/kr/lucypark ourselves. Not sure if it's worth it.

**Comment by lorepirri** [comment]
> @lorepirri They don't provide any info on where they got the jars from. SOMe of them kind of seem related to https://github.com/koalanlp/koalanlp but were committed 10 years ago which doesn't match with what I found on [maven central](https://central.sonatype.com/).
> 
> The only thing we could do is compile https://github.com/konlpy/konlpy/tree/v0.6.0/konlpy/java/src/kr/lucypark ourselves. Not sure if it's worth it.

Thank you @JeanChristopheMorinPerso , noted. I will ask SF also, because they will go to their channel.

**Comment by msentissi** [comment]
Closing because of the presence of JAR files.

**Pull request closed without merging**